engine/src/main/java/com/google/android/fhir/sync/FhirDataStore.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.sync
+
+import android.content.Context
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import androidx.work.Data
+import androidx.work.workDataOf
+import com.google.android.fhir.OffsetDateTimeTypeAdapter
+import com.google.gson.GsonBuilder
+import java.io.IOException
+import java.time.OffsetDateTime
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.runBlocking
+import timber.log.Timber
+
+@PublishedApi
+internal class FhirDataStore(context: Context) {
+  private val Context.dataStore by
+    preferencesDataStore(
+      name = FHIR_PREFERENCES_NAME,
+    )
+  private val dataStore = context.dataStore
+  private val gson =
```

Comment:
can we have an intenral data class to encapsulate the serialization and deserialization of the SyncJobStatuses?

```
internal class SyncJobStatusSerializer {
    private val gson = GsonBuilder()
        .registerTypeAdapter(OffsetDateTime::class.java, OffsetDateTimeTypeAdapter().nullSafe())
        .setExclusionStrategies(FhirSyncWorker.StateExclusionStrategy())
        .create()

    fun deserialize(data: String): SyncJobStatus? {
        return gson.fromJson(data, SyncJobStatus::class.java)
    }

    fun serialize(syncJobStatus: SyncJobStatus?): String {
        return gson.toJson(syncJobStatus)
    }
}
```

that we initialize here:
```suggestion
   private val serializer = SyncJobStatusSerializer()
```

Reply:
done.